### PR TITLE
Add SRBMiner (AMD)

### PR DESCRIPTION
**Pull request by UselessGuru**
https://bitcointalk.org/index.php?topic=3167363.0

### CODE REVIEW

**Comment by grantemsley** [comment]
Unfortunately I don't have any working AMD cards at the moment to test this with.

**Comment by angelics** [comment]
wrong class in api, **dstm**. no amd to test too

File: MinersLegacy/SRBMiner.ps1 [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        $ConfigFile = "$($Miner_Name)-$($Algorithm)-Port$($Miner_Port).json"
+      
+        (
+            [PSCustomObject]@{
+                api_enabled      = $true
+                api_port         = [Int]$Miner_Port
+                api_rig_name     = "$($Config.Pools.$($Pools.$Algorithm_Norm.Name).Worker)"
+                cryptonight_type = $Algorithm
+                gpu_conf         = @([PSCustomObject]@{
+                    "id"        = [Int]$Miner_Device.PlatformId_Index
```
Comment (UselessGuru) [inline MinersLegacy/SRBMiner.ps1]: @Aaron: Is this now the correct index?

**Comment by Kettoch** [comment]
Well, it worked on my AMD system, but I get an error:
Failed to connect to miner (SRBMiner-2)

File: MinersLegacy/SRBMiner.ps1 (hunk 1 above)
Comment (aaronsace) [reply MinersLegacy/SRBMiner.ps1]: Yes.

Comment (aaronsace) [reply MinersLegacy/SRBMiner.ps1]: No, sorry, this is completely wrong. You are converting an 'array' to an 'int' plus this miner does not appear to use platforms (or at least you haven't set one).

**Comment by Kettoch** [comment]
That is an interesting thought.  I have a fully configured config file.  Never thought to change the start.bat.

I will experiment with this.  I know this is the obvious effect of using the config file.  But until more recently I have been slow to make changes until I understood the workings of the system.

File: MinersLegacy/SRBMiner.ps1 [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        $ConfigFile = "$($Miner_Name)-$($Algorithm)-Port$($Miner_Port).json"
+      
+        (
+            [PSCustomObject]@{
+                api_enabled      = $true
+                api_port         = [Int]$Miner_Port
+                api_rig_name     = "$($Config.Pools.$($Pools.$Algorithm_Norm.Name).Worker)"
+                cryptonight_type = $Algorithm
+                gpu_conf         = @([PSCustomObject]@{
```
Comment (UselessGuru) [inline MinersLegacy/SRBMiner.ps1]: Changed to Type_PlatformId_Index

File: MinersLegacy/SRBMiner.ps1 (hunk 1 above)
Comment (UselessGuru) [reply MinersLegacy/SRBMiner.ps1]: very true... changed.

**Comment by UselessGuru** [comment]
> Can you go fully command line or fully config file?
> That would mean you could get rid of the file writes inside the miner file.

No, not to my understanding of the miner documentation.
The config file is always required. A separate pool file could be done, that would mean TWO files, and we'd still need some command line parameters.

**Comment by Kettoch** [comment]
Oh, I thought you meant the main config file.

**Comment by UselessGuru** [comment]
@aaronsace: Why do you want to write the pools.txt file? All required information can be passed by command line? This saves writing another file.

**Pull request closed without merging**